### PR TITLE
issue-2265: Fixes setMetadata for tf keys

### PR DESCRIPTION
**Pull request by friedlou**
issue-2265: Unit test needed update due to metadata change in the attribute

### CODE REVIEW

File: warehouse/query-core/src/main/java/datawave/query/attributes/Attribute.java [hunk 1]
```diff
@@ -117,18 +117,18 @@ protected void setMetadata(Key key) {
             } else if (isTermFrequency(cf)) {
                 // find the second null byte in the cq and take everything before that (cq = DataType\0UID\0Normalized Field Value\0Field Name)
                 final ByteSequence cq = key.getColumnQualifierData();
-                int nullOffset = 0;
+                int length = cq.length();
                 int count = 0;
-                for (int i = 0; i < cf.length(); i++) {
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/attributes/Attribute.java:122]: Wow, this was really wrong.  Good find!

**Review by ivakegg — APPROVED** [review]
Good job!

**Review by avgAGB — APPROVED** [review]

**Comment by apmoriarty** [comment]
Alternatively, https://github.com/NationalSecurityAgency/datawave/pull/2253

**Review by apmoriarty — APPROVED** [review]

**Comment by friedlou** [comment]
Folded int a different PR

**Pull request closed without merging**